algoliasearch/src/main/java/com/algolia/search/saas/BaseIndex.java

```diff
@@ -342,7 +342,20 @@ protected JSONObject search(Query query) throws AlgoliaException {
     }
 
     /**
-     * Wait the publication of a task on the server. 
+     * Search inside the index
+     * @return a byte array containing search results
+     * @throws AlgoliaException
+     */
+    protected byte[] searchRaw(Query query) throws AlgoliaException {
+        String paramsString = query.getQueryString();
+        if (paramsString.length() > 0)
```

Comment:
A code cleanup is planned to cope with these :wink: 

Still, done for this file (Query.java has **55** `if`s without braces !), I'll clean the rest afterwards :smile: